### ISSUE TITLE
How to open the TIFF format acquired by JEM-2100

### ISSUE THREAD

**Issue by PhysXu** [hyperspy/hyperspy#2511]
I want to open some data acquired by the instrument of JEM-2100, in the version of TIFF, but have some unknown problems as follow:
![image](https://user-images.githubusercontent.com/35793078/91695210-629d6b80-eba0-11ea-937c-fb72db4622cb.png)

BTW, the version of my HyperSpy is 1.52.
[30592.zip](https://github.com/hyperspy/hyperspy/files/5149000/30592.zip)

**Comment by tjof2** [comment]
Here's the metadata according to ImageJ.

It looks like you have a stack of two images, one is 8-bit and that includes the scalebar, while the other is 16-bit and doesn't include the scalebar.

The 8-bit image with the scalebar seems like it's meant to be RGB, since `SizeC = 3`, and indeed ImageJ displays it as such. But `IsRGB = false`, so it looks like it's badly implemented.
```
 30592.tif #1 BitsPerPixel = 8
 30592.tif #1 DimensionOrder = XYCZT
 30592.tif #1 IsInterleaved = false
 30592.tif #1 IsRGB = false
 30592.tif #1 LittleEndian = true
 30592.tif #1 PixelType = uint8
 30592.tif #1 SizeC = 3
 30592.tif #1 SizeT = 1
 30592.tif #1 SizeX = 4032
 30592.tif #1 SizeY = 2688
 30592.tif #1 SizeZ = 1
 30592.tif #2 BitsPerPixel = 16
 30592.tif #2 DimensionOrder = XYCZT
 30592.tif #2 IsInterleaved = false
 30592.tif #2 IsRGB = false
 30592.tif #2 LittleEndian = true
 30592.tif #2 PixelType = uint16
 30592.tif #2 SizeC = 1
 30592.tif #2 SizeT = 1
 30592.tif #2 SizeX = 4032
 30592.tif #2 SizeY = 2688
 30592.tif #2 SizeZ = 1
 Series 0 Name = 30592.tif #1
 Series 1 Name = 30592.tif #2
Artist = THU
BIG_TIFF = false
BitsPerSample = 8
COLOR_SPACE = 1
Compression = JPEG
DATE_TIME_DIGITIZED = 2018:11:15 19:42:33
DateTime = 2018:11:15 19:42:31
EXPOSURE_MODE = 1
EXPOSURE_TIME = 89922/244957
ImageLength = 2688
ImageWidth = 4032
Instrument Make = Olympus Soft Imaging Solutions
Instrument Model = Quemesa
LITTLE_ENDIAN = true
METERING_MODE = 1
MetaDataPhotometricInterpretation = RGB
MetaMorph = no
NumberOfChannels = 3
Orientation = 1st row - top; 1st column - left
PIXEL_X_DIMENSION = 4032
PIXEL_Y_DIMENSION = 2688
PhotometricInterpretation = RGB
PlanarConfiguration = Chunky
ResolutionUnit = Inch
SamplesPerPixel = 3
XResolution = 72.0
YCbCrSubSampling = chroma image dimensions = luma image dimensions
YResolution = 72.0
```
As @ericpre suggested on Github, do you have the [`imagecodecs`](https://pypi.org/project/imagecodecs/) package installed?

In any case, it might be that your image falls into one of the unsupported categories of [`tifffile`](https://pypi.org/project/tifffile/):
> A subset of the TIFF specification is supported, mainly uncompressed and losslessly compressed 8, 16, 32 and 64-bit integer, 16, 32 and 64-bit float, grayscale and multi-sample images. **Specifically, reading slices of image data, CCITT and OJPEG compression, chroma subsampling without JPEG compression, color space transformations, samples with differing types, or IPTC and XMP metadata are not implemented.**

**Comment by ericpre** [comment]
It works fine with the latest release and this may due to the fact that the the tifffile library is vendored in 1.5.2 is vendored (#2287) and most likely doesn't support the JPEG codecs through `imagecodecs` or alternative.

**Comment by PhysXu** [comment]
> It works fine with the latest release and this may due to the fact that the the tifffile library is vendored in 1.5.2 is vendored (#2287) and most likely doesn't support the JPEG codecs through `imagecodecs` or alternative.

Thanks for your help. It is fixed after I install the **imagecodecs** package

**Comment by PhysXu** [comment]
![image](https://user-images.githubusercontent.com/35793078/91713526-87074100-ebbc-11ea-88b3-c81c3706faed.png)
additional question:
The information of scalebar seems to lose when I opened it directly by using HyperSpy.

BTW, I have installed **imagecodecs** package and updated my HyperSpy by Conda.

**Comment by ericpre** [comment]
I am not sure that the calibration is actually saved in the file, fairly likely that the file that the file was saved/exported with the wrong option. @PhysXu, how do you know that the calibration is saved in the file?

**Comment by PhysXu** [comment]
@ericpre 
The RADIUS software told me that the TIFF has been calibrated well, seeing below:
![image](https://user-images.githubusercontent.com/35793078/91715035-82905780-ebbf-11ea-90bf-34d48da4d9ea.png)

This is a licensed-software, so that I cannot use it in other computers and turn to try HyperSpy.

**Comment by ericpre** [comment]
The only way forward is for you to figure out where is the calibration is saved. As mentioned above, in your situation, I would start with checking to save/export the data. In the metadata, it doesn't seem that there is any suitable calibration data:
```python
s.original_metadata
├── Artist = THU
├── BitsPerSample = (8, 8, 8)
├── Compression = <COMPRESSION.JPEG: 7>
├── DateTime = 2018:11:15 19:42:31
├── ExifTag
│   ├── ColorSpace = 1
│   ├── DateTimeDigitized = 2018:11:15 19:42:33
│   ├── ExifVersion = 0210
│   ├── ExposureMode = 1
│   ├── ExposureTime = (89922, 244957)
│   ├── FlashpixVersion = 0100
│   ├── MeteringMode = 1
│   ├── PixelXDimension = 4032
│   └── PixelYDimension = 2688
├── ImageLength = 2688
├── ImageWidth = 4032
├── JPEGTables = b'\xff\xd8\xff\xdb\x00C\x00\x05\x03\x04\x04\x04\x03\x05\x04\x04\x04\x05\x05\x0 ... 3\xe4\xe5\xe6\xe7\xe8\xe9\xea\xf1\xf2\xf3\xf4\xf5\xf6\xf7\xf8\xf9\xfa\xff\xd9'
├── Make = Olympus Soft Imaging Solutions
├── Model = Quemesa
├── Orientation = <ORIENTATION.TOPLEFT: 1>
├── PhotometricInterpretation = <PHOTOMETRIC.RGB: 2>
├── PlanarConfiguration = <PLANARCONFIG.CONTIG: 1>
├── ResolutionUnit = <RESUNIT.INCH: 2>
├── RowsPerStrip = 2688
├── SamplesPerPixel = 3
├── StripByteCounts = (8206089,)
├── StripOffsets = (409,)
├── XResolution = (72, 1)
├── YCbCrSubSampling = (1, 1)
└── YResolution = (72, 1)
```

**Comment by tjof2** [comment]
@ericpre  does the 16-bit image that is also in the file have any calibration? It's a stack remember, and you've posted the 8-bit metadata only I think

**Comment by cgohlke** [comment]
> The information of scalebar seems to lose when I opened it directly by using HyperSpy.

The physical size is buried in a OlympusSIS tag in the second page:
```Python
>>> import tifffile
>>> tifffile.TiffFile('30592.tif').pages[1].tags[33560].value
{'name': '', 'datetime': datetime.datetime(2018, 11, 15, 19, 42), 'pixelsizex': 2.392791775876109e-11, 'pixelsizey': 2.3927917758761087e-11, 'magnification': 600000.0, 'cameraname': '', 'picturetype': ''}
```

**Comment by tjof2** [comment]
So the question now is: is this a standard format that HyperSpy should support, or is it non-standard and it can be closed?

**Comment by thomasaarholt** [comment]
Specifically which software is this saved with, @PhysXu?

**Comment by ericpre** [comment]
Thanks @cgohlke for the pointer, this should be fixed in #2525.

**Comment by tjof2** [comment]
Close in #2525
